discord_bots/views/configure_category.py

```diff
@@ -0,0 +1,319 @@
+        except Exception:
+            await interaction.response.send_message(
+                embed=Embed(
+                    description="Decay amount & proportion must be a decimal number",
+                    colour=Colour.red(),
+                ),
+                ephemeral=True,
+            )
+            return
+
+        try:
+            if not str(self.sigma_decay_grace_days) == "":
+                self.decay_grace_days = int(str(self.sigma_decay_grace_days))
+        except Exception:
+            await interaction.response.send_message(
+                embed=Embed(
+                    description="Decay grace days must be an integer",
+                    colour=Colour.red(),
+                ),
+                ephemeral=True,
+            )
+            return
+
+        # Check for zero's
+        if (
+            (self.decay_amount and self.decay_amount < 0)
+            or (self.decay_grace_days and self.decay_grace_days < 0)
+            or (self.decay_max_decay_proportion and self.decay_max_decay_proportion < 0)
+        ):
+            await interaction.response.send_message(
+                embed=Embed(
+                    description="Value must be greater than or equal to 0",
+                    colour=Colour.red(),
+                ),
+                ephemeral=True,
+            )
+            return
+
+        if self.decay_amount:
+            self.view.category.sigma_decay_amount = self.decay_amount
+            if self.view.embed.description:
+                self.view.embed.description = (
+                    self.view.embed.description
+                    + f"\nSigma decay amount: **{self.view.category.sigma_decay_amount}**"
+                )
+
+        if self.decay_grace_days:
+            self.view.category.sigma_decay_grace_days = self.decay_grace_days
+            if self.view.embed.description:
+                self.view.embed.description = (
+                    self.view.embed.description
+                    + f"\nSigma decay grace days: **{self.view.category.sigma_decay_grace_days}**"
+                )
+
+        if self.decay_max_decay_proportion:
+            self.view.category.sigma_decay_max_decay_proportion = (
+                self.decay_max_decay_proportion
+            )
+            if self.view.embed.description:
+                self.view.embed.description = (
+                    self.view.embed.description
+                    + f"\nSigma max decay proportion: **{self.view.category.sigma_decay_max_decay_proportion}**"
+                )
+
+        await self.view.interaction.edit_original_response(embed=self.view.embed)
+
+        # Interaction must be responded to, but is then deleted
+        await interaction.response.send_message(
+            embed=Embed(
+                description=f"Set decay settings successful",
+                colour=Colour.blue(),
+            ),
+            ephemeral=True,
+        )
+        await interaction.delete_original_response()
```

Comment:
I'm confused here without having played around with the modals: You are editing the original response, sending a new response, and then deleting the original response. Shouldn't the first edit be enough, since that is counted as a response to the interaction?